README.md

```diff
@@ -8,61 +8,131 @@ tailored to simplify the development of Configuration as Code tools like Monaco,
+  | Classic config APIs      | ❌           |
+  | Settings 2.0             | ❌           |
+  | Settings 2.0 permissions | ✅           |
+  | Automation               | ✅           |
+  | Grail buckets            | ✅           |
+  | Documents                | ✅           |
+  | OpenPipeline             | ✅           |
+  | Segments                 | ✅           |
+  | SLO's                    | ✅           |
+  | Account management       | ✅           |
 
 ### Usage
+To instantiate a client, it's recommended to create an instance via the provided `clients.Factory()` function.
 
-To instantiate an API client, it's recommended to create an instance via the provided `clients.Factory()` function:
+#### Platform clients
+Platform clients are designed to interact with Dynatrace platform APIs.
 
+Ensure that you are using the correct environment URL, which must include `.apps.dynatrace.com`.
+Authentication can be handled using either OAuth or a platform token`.
 ```go
 // create the factory
+ctx := context.TODO()
 factory := clients.Factory().
-	WithEnvironmentURL("https://dt-environment.com").
+    WithEnvironmentURL("https://<dt-environment>.apps.dynatrace.com").
 	WithOAuthCredentials(credentials)
+    // or if you want to use a platform token
+    WithPlatformToken("<YOUR_PLATFORM_TOKEN>")
 
-// request any client from the factory, e.g. bucket api client
-bucketClient, err := factory.BucketClient()
+// request any client from the factory, e.g. bucket client
+bucketClient, err := factory.BucketClient(ctx)
 if err != nil {
 	// handle error
 }
 
 // perform operation
-resp, err := bucketClient.Get(context.TODO(), "my bucket")
+resp, err := bucketClient.Get(ctx, "my bucket")
+
+if err != nil {
+    // handle error. See Error handling section.
+}
+
+// unmarshal payload
+bucketDefinition, err := api.DecodeJSON[BucketDefinition](resp.Response)
 if err != nil {
 	// handle error
 }
+```
+
+#### Classic rest client
+Unlike [Platform clients](#platform-clients), classic clients do not include dedicated resource clients.
+Instead, only a general-purpose REST client is available for interacting with the API.
 
-// inspect response
-if !resp.IsSuccess() {
-	// handle api error
+```go
+// create the factory
+ctx := context.TODO()
+factory := clients.Factory().
+    WithClassicURL("https://<dt-environment>.live.dynatrace.com").
+    WithAccessToken("<YOUR_ACCESS_TOKEN>")
+
+// create a classic client from the factory
+client, err := cFactory.CreateClassicClient()
+if err != nil {
+	// handle error
+}
+
+// perform operation
+httpResp, err := client.GET(ctx, "/your-endpoint", rest.RequestOptions{})
+
+if err != nil {
+    // handle client error
+}
+
+resp, err := api.NewResponseFromHTTPResponse(resp)
+
+if err != nil {
+    // handle error. See Error handling section.
 }
 
 // unmarshal payload
-bucketDefinition, err := api.DecodeJSON[BucketDefinition](resp.Response)
+data, err := api.DecodeJSON[YourExpectedStruct](resp.Response)
```

Comment:
We almost never use this function `api.DecodeJSON`... perhaps we consider this as a work item for the future: use or remove.